### PR TITLE
Small PIDTransientControl enhancement

### LINKED ISSUE #26742: PIDTransientControl object enhancement
## Reason
PIDTransientControl checks if the executioner is Transient and outputs a mooseError if it isn't. However, at least one moose based app has executioners that are "quasi" transients but inherit from Eigenvalue instead of Transient since inheriting from both is not possible. The PIDTransientControl can be used with these executioners if the check is changed to `if (!_fe_problem.isTransient())`. Furthermore, PIDTransientControl would be more useful with two parameters that allow a user to limit the output value between a minimum and maximum value.

## Design
Update the check in the constructor to `if (!_fe_problem.isTransient())` and add two new parameters.

## Impact
PIDTransientControl will be more useful in Griffin and potentially other moose based apps.

### PR DESCRIPTION
**Pull request by jthano**
closes #26742

### CODE REVIEW

File: test/tests/controls/pid_control/tests [hunk 1]
```diff
@@ -8,8 +8,19 @@
...
+      # Not ouputting left_boundary_average to avoid regold
+      cli_args = 'Postprocessors/left_boundary_average/outputs=none'
       detail = 'using the principle of Proportional Integral Derivative control,'
     []
+    [min_max_limits]
+      type = 'CSVDiff'
+      input = 'pid_control.i'
+      csvdiff = 'min_max_limits.csv'
+      recover = false
+      # Not ouputting left_boundary_average to avoid regold
```
Comment (GiudGiud) [inline test/tests/controls/pid_control/tests:20]:
```suggestion
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
There's two changes here, but only one is tested

Add to documentation:

# How to use in a transient
...

# How to use in a steady solve
...
!listing

Add an input for steady solves if you want to enable this

**Comment by jthano** [comment]
> There's two changes here, but only one is tested
> 
> Add to documentation:
> 
> # How to use in a transient
> ...
> 
> # How to use in a steady solve
> ... !listing
> 
> Add an input for steady solves if you want to enable this

Okay, I'll look into this more. I mentioned `Steady` executioner in the warning message because the original mooseError message said `If using a Steady Executioner, add a minimum number of Picard iterations and comment this error.` Do you know what the use case was that inspired the original message? I could just change my `mooseWarning` to a `mooseError` and not mention the Steady Executioner?

**Comment by GiudGiud** [comment]
> Do you know what the use case was that inspired the original message? I

the use case is that you can consider doing a picard multiapp iteration with the PIDController active. 
Didnt it start as a mooseError? Are you planning to use this with a steady executioner

**Comment by jthano** [comment]
> > Do you know what the use case was that inspired the original message? I
> 
> the use case is that you can consider doing a picard multiapp iteration with the PIDController active. Didnt it start as a mooseError? Are you planning to use this with a steady executioner

No in my use case the problem will be a transient (with a custom Executioner that is not Transient) and I won't see this warning message. I didn't understand the original `mooseError` message that well. It seems that if you suggest to the user to comment out an error and recompile for their use case, then it should be a warning, as in the usage they want is advanced and non-typical but supported to some extent. I can add a test for using this with a `Steady` executioner or I can just change my `mooseWarning` back to a `mooseError` with a message similar to the original one. I like the idea of trying to add a test with a `Steady` executioner if the original intention of the object was to support that as an advanced use case.

**Comment by GiudGiud** [comment]
I did not have time to make sure it worked with Steady so I added an error instead of a warning. 
It's not super supported. The P ok, the Integral and the derivative are meh  / ill-defined with a steady executioner.

**Comment by jthano** [comment]
> I did not have time to make sure it worked with Steady so I added an error instead of a warning. It's not super supported. The P ok, the Integral and the derivative are meh / ill-defined with a steady executioner.

Okay, I'll change it back to a `mooseError` then. I think that is the best solution.

**Comment by jthano** [comment]
I changed the warning to an error. Let me know if you agree with the message. I think if someone tries to use the object with a Steady executioner, they will know based on the error message to open an issue and ask for that to be supported or do it themselves.

**Review by GiudGiud — APPROVED** [review]
looks good.

File: test/tests/controls/pid_control/tests (hunk 1 above)
Comment (jthano) [reply test/tests/controls/pid_control/tests]: removed the comment in question and fixed the typo in the other two comments I added

**Pull request merged**